src/app/components-small/plugin-filter-node/plugin-filter-node.component.ts

```diff
@@ -36,47 +39,47 @@ export class PluginFilterNodeComponent implements OnInit {
             console.warn("No filter object provided to plugin filter node component");
             return;
         }
-        const isLeaf = (filter.name != null) || filter.tag || filter.version;
+        // Should filters have multiple attributes at some point, this must be changed
+        const type = Object.keys(filter)[0];
+        if (type !== 'and' && type !== 'or' && type !== 'name' && type !== 'tag' && type !== 'version') {
```

Comment:
Create a static set of allowed filter types and test if type is in that set (or refactor this test out into a method).
Even if this is probably the most optimized version it is also hard to read and maintain.